Automattic/wp-calypso
# Checkout: Convert smaller composite checkout dependencies to TypeScript

#### Changes proposed in this Pull Request

This is a follow-up to https://github.com/Automattic/wp-calypso/pull/46758 which converts the main `CompositeCheckout` component to TypeScript.

In this PR we convert some of the smaller dependencies of `CompositeCheckout` to TypeScript as well, specifically:

- useIsApplePayAvailable
- filterAppropriatePaymentMethods
- useStoredCards
- useCachedDomainContactDetails
- useCountryList

#### Testing instructions

- Make sure Apple Pay loads when it should (testing instructions are beyond the scope of this PR).
- Make sure you see the payment methods you expect when paying with or without full credits.
- Make sure that stored cards are listed as payment methods.
- Make sure that cached contact details load as expected.
- Make sure that all the counties you'd expect are listed in the contact details form.
